java/aquality-selenium-template/src/test/java/aquality/selenium/template/glue/hooks/ScreenshotHooks.java

```diff
@@ -0,0 +1,39 @@
+package aquality.selenium.template.glue.hooks;
+
+import aquality.selenium.browser.AqualityServices;
+import io.cucumber.core.api.Scenario;
+import io.cucumber.java.After;
+import ru.yandex.qatools.ashot.AShot;
+import ru.yandex.qatools.ashot.Screenshot;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategies;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
+
+import javax.imageio.ImageIO;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class ScreenshotHooks {
+
+    @After(order = 1)
+    public void takeScreenshot(Scenario scenario) {
+        if (scenario.isFailed()) {
+            scenario.embed(attachScreenshot(), "image/png");
+        }
+    }
+
+    private byte[] attachScreenshot() {
+        int scrollTimeout = 500;
+        ShootingStrategy shootingStrategy = ShootingStrategies.viewportPasting(scrollTimeout);
+        Screenshot fpScreenshot = new AShot()
+                .shootingStrategy(shootingStrategy)
+                .takeScreenshot(AqualityServices.getBrowser().getDriver());
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            ImageIO.write(fpScreenshot.getImage(), "jpg", baos);
+            baos.flush();
+            return baos.toByteArray();
+        } catch (IOException ioe) {
+            AqualityServices.getLogger().warn(String.format("IO Exception during preparing screenshot of full page%nException message: %s", ioe.getMessage()));
```

Comment:
shouldn't we pass exception to logger by our coding guidelines?

Reply:
We have only fatal log with exception. Do we really need it?

Reply:
seems that debug should also have it. 
I'm not sure, but seems that we should show a good code as a template, an as I can remember, logging of exception itself rather than only a message was strictly recommended previously

Reply:
fixed